src/java.base/share/classes/java/nio/Direct-X-Buffer.java.template

```diff
@@ -239,6 +240,19 @@ class Direct$Type$Buffer$RW$$BO$
 #end[rw]
     }
 
+    // Constrain the capacity to int range throwing IAE if not possible
+    //
+    private static int clampCapacity(long capacity) {
+        try {
+            return Math.toIntExact(capacity);
```

Comment:
Why not the more direct and obvious:
```
if (capacity < 0) {
  throw ...
} else if (capacity > Integer.MAX_VALUE) {
  throw ...
}
```

Reply:
So changed in d2386a4b73f0fa4eb9cc278b3cbb1578e6cb4b55.